File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-2.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode a = list1;
        ListNode b = list2;
        ListNode dummy = new ListNode(0);
        ListNode current = dummy;
        while(a!=null && b!=null){
            if(a.val<=b.val){
                current.next=a;
                a=a.next;
            }
            else{
                current.next=b;
                b=b.next;
            }
            current=current.next;
        }
        if(a!=null){
            current.next=a;
        }
        if(b!=null){
            current.next=b;
        } 
        return dummy.next;
    }
}

File: Data Structures & Algorithms/reverse-nodes-in-k-group/submission-2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reverseKGroup(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if head is None:
            return head

        temp = head
        cnt = 0
        while temp is not None:
            cnt+=1
            temp = temp.next

        if cnt<k:
            return head
        
        prev = None
        current = head
        next = None
        count = 0

        while current is not None and count<k:
            next = current.next
            current.next = prev
            prev =current
            current = next
            count+=1
        
        if next is not None:
            head.next = self.reverseKGroup(next, k)

        return prev
        
        